### PR TITLE
Add custom log source support

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?

This PR adds support to setting the log source via resource attribute OR log attribute `datadog.log.source`. If this is not set, it falls back to the previous behaviour (hardcoded log source).

This change is applied in all three paths:
- OTLP ingest in agent
- Datadog Exporter
- Converged agent

### Describe how to test/QA your changes
**Resource attribute:**
Set resource attribute datadog.log.source and send logs. Ensure the source is being populated in the app.

**Log attribute:** 
Set log attribute datadog.log.source and send logs. Ensure the source is being populated in the app.

You can use the following app to emit the logs:
```
package main

import (
	"context"
	"fmt"
	"time"

	// "go.opentelemetry.io/otel/attribute"
	"go.opentelemetry.io/otel/attribute"
	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp"
	logapi "go.opentelemetry.io/otel/log"
	"go.opentelemetry.io/otel/log/global"
	"go.opentelemetry.io/otel/sdk/log"
	"go.opentelemetry.io/otel/sdk/resource"
	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
)

func main() {
	ctx := context.Background()

	// Create resource.
	res, err := newResource()
	if err != nil {
		panic(err)
	}

	// Create a logger provider.
	// You can pass this instance directly when creating bridges.
	loggerProvider, err := newLoggerProvider(ctx, res)
	if err != nil {
		panic(err)
	}

	// Handle shutdown properly so nothing leaks.
	defer func() {
		if err := loggerProvider.Shutdown(ctx); err != nil {
			fmt.Println(err)
		}
	}()

	global.SetLoggerProvider(loggerProvider)
	logger := loggerProvider.Logger("otel_logger")
	record := logapi.Record{}
	// record.AddAttributes(logapi.String("datadog.log.source", "customsrc"))

	record.SetBody(logapi.StringValue("THIS IS THE BODY"))
	for {
		logger.Emit(context.Background(), record)
		time.Sleep(5 * time.Second)
	}
}

func newResource() (*resource.Resource, error) {
	return resource.Merge(resource.Default(),
		resource.NewWithAttributes(semconv.SchemaURL,
			semconv.ServiceName("my-service"),
			semconv.ServiceVersion("0.1.0"),
			attribute.String("datadog.log.source", "customsrc"),
		))
}

func newLoggerProvider(ctx context.Context, res *resource.Resource) (*log.LoggerProvider, error) {
	exporter, err := otlploghttp.New(ctx, otlploghttp.WithInsecure())
	if err != nil {
		return nil, err
	}
	processor := log.NewBatchProcessor(exporter)
	provider := log.NewLoggerProvider(
		log.WithResource(res),
		log.WithProcessor(processor),
	)
	return provider, nil
}

```

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]

File: comp/otelcol/otlp/components/exporter/logsagentexporter/logs_exporter_test.go [hunk 1]
```diff
@@ -72,17 +72,19 @@ func TestLogsExporter(t *testing.T) {
 					lrr := testutil.GenerateLogsOneLogRecord()
 					ldd := lrr.ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().At(0)
 					ldd.Attributes().PutStr("message", "hello")
+					ldd.Attributes().PutStr("datadog.log.source", "custom_source")
```
Comment (songy23) [inline comp/otelcol/otlp/components/exporter/logsagentexporter/logs_exporter_test.go:75]: Can you add another test that sets this attribute in resource attributes?

Comment (mackjmr) [reply comp/otelcol/otlp/components/exporter/logsagentexporter/logs_exporter_test.go:75]: Added in last commit 👍

File: comp/otelcol/otlp/components/exporter/logsagentexporter/logs_exporter_test.go [hunk 2]
```diff
@@ -99,6 +99,40 @@ func TestLogsExporter(t *testing.T) {
...
+			args: args{
+				ld: func() plog.Logs {
+					l := testutil.GenerateLogsOneLogRecord()
+					rl := l.ResourceLogs().At(0)
+					resourceAttrs := rl.Resource().Attributes()
+					resourceAttrs.PutStr("datadog.log.source", "custom_source_rattr")
+					return l
+				}(),
+				otelSource:    otelSource,
+				logSourceName: "custom_source_rattr",
```
Comment (songy23) [inline comp/otelcol/otlp/components/exporter/logsagentexporter/logs_exporter_test.go:113]: It looks like setting this field forces log source to be `custom_source_rattr`, regardless of the attribute `datadog.log.source`, so this test may not really validate if `datadog.log.source` works

https://github.com/DataDog/datadog-agent/blob/fe16b1d2d98595695e79f269fb1db68198771544/comp/otelcol/otlp/components/exporter/logsagentexporter/logs_exporter_test.go#L267

Suggest to change `logSourceName` back to `otlp_log_ingestion` and then see if `datadog.log.source` takes precedence.  Same for the other test.

Comment (mackjmr) [reply comp/otelcol/otlp/components/exporter/logsagentexporter/logs_exporter_test.go]: Good call, updated the test 👍

**Review by songy23 — APPROVED** [review]
LGTM

**Pull request merged**